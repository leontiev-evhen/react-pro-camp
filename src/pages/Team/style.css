.team-section h1 {
  font-size: 30px;
  font-weight: bold;
}

.team-section figure {
  text-align: center;
}

.team-section li {
  list-style-type: none;
}