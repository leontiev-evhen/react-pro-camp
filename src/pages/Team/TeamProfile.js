import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col, Figure } from 'react-bootstrap';
import './style.css';

function TeamProfile({ team }) {
	return (
		<div className="team-section">
			<Row>
				<Col md={3}>
					<Figure>
						<Figure.Image width={171} height={180} alt="logo" src={team.logo} />
						<Figure.Caption>
							<div>{team.venue_city}</div>
							<div>{team.venue_address}</div>
						</Figure.Caption>
					</Figure>
				</Col>
				<Col md={9}>
					<h1>{team.name}</h1>
					<ul>
						<li>
							<b>Country:</b> {team.country}
						</li>
						<li>
							<b>Founded:</b> {team.founded}
						</li>
						<li>
							<b>Stadium:</b> «{team.venue_name}»
						</li>
						<li>
							<b>Capacity:</b> {team.venue_capacity}
						</li>
					</ul>
				</Col>
			</Row>
		</div>
	);
}

TeamProfile.propTypes = {
	team: PropTypes.shape({
		logo: PropTypes.string,
		venue_city: PropTypes.string,
		venue_address: PropTypes.string,
		name: PropTypes.string,
		country: PropTypes.string,
		founded: PropTypes.number,
		venue_name: PropTypes.string,
		venue_capacity: PropTypes.number,
	}).isRequired,
};

export default TeamProfile;
