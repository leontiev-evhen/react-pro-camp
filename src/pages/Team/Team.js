import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ContainerHOC from '../../hoc/ContainerHOC';
import { getTeamById } from '../../providers/teams';
import TeamProfile from './TeamProfile';

const DivContainerHOC = ContainerHOC(TeamProfile);

class Team extends Component {
	state = {
		team: {},
		isFetching: true,
		error: '',
	};

	componentDidMount() {
		this.fetchTeam();
	}

	fetchTeam = async () => {
		const id = this.props.match.params.id;
		try {
			const response = await getTeamById(id);
			this.setState({
				team: response.api.teams[0],
				isFetching: false,
			});
		} catch (e) {
			this.setState({
				error: e.message || e.data.message,
				isFetching: false,
			});
		}
	};
	render() {
		const { team, isFetching, error } = this.state;
		return (
			<DivContainerHOC team={team} isFetching={isFetching} error={error} />
		);
	}
}

Team.propTypes = {
	match: PropTypes.object.isRequired,
};

export default Team;
