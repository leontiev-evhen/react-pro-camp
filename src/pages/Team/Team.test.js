import React from 'react';
import { shallow } from 'enzyme';
import { Team } from './Team';

describe('Team component', () => {
	const props = {
		match: {
			params: { id: 1 },
		},
	};

	const Component = shallow(<Team {...props} />);

	it('fetchTeam should be called', () => {
		const instance = Component.instance();
		jest.spyOn(instance, 'fetchTeam');
		instance.componentDidMount();
		expect(instance.fetchTeam).toHaveBeenCalledTimes(1);
	});

	it('Team component render correctly', () => {
		expect(Component).toMatchSnapshot();
	});
});
