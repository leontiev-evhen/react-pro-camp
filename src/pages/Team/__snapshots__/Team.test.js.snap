// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team component Team component render correctly 1`] = `
<containerHOC
  error=""
  isFetching={true}
  team={Object {}}
/>
`;
